addons/payment_xendit/static/src/js/payment_form.js

```diff
@@ -0,0 +1,205 @@
+
+    // #=== PAYMENT FLOW ===#
+
+    /**
+     * Trigger the payment processing by submitting the data.
+     *
+     * @override method from payment.payment_form
+     * @private
+     * @param {string} providerCode - The code of the selected payment option's provider.
+     * @param {number} paymentOptionId - The id of the selected payment option.
+     * @param {string} paymentMethodCode - The code of the selected payment method, if any.
+     * @param {string} flow - The payment flow of the selected payment option.
+     * @return {void}
+     */
+    async _initiatePaymentFlow(providerCode, paymentOptionId, paymentMethodCode, flow) {
+        if (providerCode !== 'xendit' || flow === 'token' || paymentMethodCode != 'card') {
+            this._super(...arguments); // Tokens are handled by the generic flow
+            return;
+        }
+        const formInputs = this._xenditGetInlineFormInputs(paymentOptionId, paymentMethodCode)
+        const details = this._xenditGetPaymentDetails(paymentOptionId, paymentMethodCode)
+
+        Object.keys(formInputs).forEach(el => formInputs[el].setCustomValidity(""))
+        if(!Xendit.card.validateCardNumber(details.card_number)){
+            formInputs['card'].setCustomValidity(_t("Invalid Card Number"))
+        }
+        if(!Xendit.card.validateExpiry(details.card_exp_month, details.card_exp_year)){
+            formInputs['month'].setCustomValidity(_t("Invalid Date"))
+            formInputs['year'].setCustomValidity(_t("Invalid Date"))
+        }
+        if(!Xendit.card.validateCvn(details.card_cvn)){
+            formInputs['cvn'].setCustomValidity(_t("Invalid CVN"))
+        }
+
+        const inputs = Object.values(formInputs);
+
+        // Checking the elements
+        if (!inputs.every(element => element.reportValidity())) {
+            this._enableButton(); // The submit button is disabled at this point, enable it
+            return;
+        }
+
+        await this._super(...arguments);
+    },
+
+    /**
+     * Process the direct payment flow.
+     *
+     * @override method from payment.payment_form
+     * @private
+     * @param {string} providerCode - The code of the selected payment option's provider.
+     * @param {number} paymentOptionId - The id of the selected payment option.
+     * @param {string} paymentMethodCode - The code of the selected payment method, if any.
+     * @param {object} processingValues - The processing values of the transaction.
+     * @return {void}
+     */
+    async _processDirectFlow(providerCode, paymentOptionId, paymentMethodCode, processingValues) {
+        if (providerCode !== 'xendit') {
+            this._super(...arguments);
+            return;
+        }
+
+        // if tokenize, use multiUse token
+        const payload = {
+            ...this._xenditGetPaymentDetails(paymentOptionId, paymentMethodCode),
+            is_multiple_use: true,
+            should_authenticate: false,
+        }
+
+        Xendit.card.createToken(
```

Comment:
It will not be allowed, I believe an error would be thrown right away while trying to create token (which I believe is already handled currently), so I suppose no adjustmends needed here?

Reply:
No need, indeed; I just wanted confirmation ;)